### ISSUE TITLE
APK download?

### ISSUE THREAD

**Issue by IzzySoft** [joelpaulkoch/schildpad#29]
Thanks for making your app available open-source! Would you mind also making its APK available for download outside the walled gardens of Google Play, so privacy-focused people (and others without access to PlayStore) can use it? You could e.g. attach it to [you relases](https://github.com/joelpaulkoch/schildpad/releases), which you'll find quite common with FOSS Android apps :wink:

Thanks in advance!

**Comment by joelpaulkoch** [comment]
Yes, thanks for the nudge.
Publishing on F-Droid is on my list but of course I can attach the APK to the releases, just have to figure out how to properly automate it.

**Comment by IzzySoft** [comment]
I gladly help you get listed at F-Droid, too (disclosure: I'm one of the maintainers there). Having the APK would allow me to list your app in my repo "immediately" (effective with the next sync, which currently would be around 7 pm UTC each day – 6 pm UTC in daylight-savings time). That would also help you getting prepared for F-Droid, where ideally you achieve "reproducible builds" (so the APK signed by you gets shipped) – which again requires your APK to be easily available; ideally attached to… you guess it :wink:

**Comment by joelpaulkoch** [comment]
I created a release with an apk and opened an issue here: https://gitlab.com/fdroid/rfp/-/issues/2400
Could you guide me in the process to publish the app on F-Droid?

**Comment by IzzySoft** [comment]
Thanks a lot! And yes, I gladly help.

Thanks for opening the RFP already. What issuebot complained about (missing Fastlane structures), my [Fastlane Cheat Sheet](https://gitlab.com/-/snippets/1895688) can help you with: short- and full-description are mandatory, an icon plus some screenshots would be great, everything else is "icing on the cake" and can be added later if you like (or skipped if you don't like).

As for the APK: unfortunately it's twice the size my repo allows, or I had added it there right away (as indicated). Let me check… Hm, as usual, the per-ABI Flutter libs are making it that size. Could you try per-ABI builds? For apps larger than 30 MB, F-Droid would ask that anyway. If I strip it down to just the armeabi-v7a and apply full compression, the APK is down below 10 MB – as I see in the original APK the Flutter libs are just stored (not compressed); taken that into the calculation, armeabi would be around 16 MB and thus still fit in. If you'd do that, I'd offer you a "Fastlane Starter Kit" in return (via PR here) so that point would be covered.

Scanning for libs inside, as expected, shows all is clean (no proprietary libs found) – so from that point it's ready for F-Droid.

So: once you've set up Fastlane and let me know, I'd trigger a rescan at F-Droid. Guess with that the "ready" label can be applied, and the next step would be someone sets up a merge request for fdroiddata. I can help you with that again then – though not being an Android or Flutter dev, I cannot do it alone but would need your help to get the build running fine :wink:

**Comment by joelpaulkoch** [comment]
Regarding fastlane, for now I just need to declare the metadata as described right?
I've done that in the latest commit in the folder /metadata/en-US.

There is no need to actually use fastlane e.g. for CI?

**Comment by IzzySoft** [comment]
> Regarding fastlane, for now I just need to declare the metadata as described right?

Yes, you can check that. I'd recommend a slight update to `full_description.txt`, though: F-Droid converts each `\n` to `<br>`, so you will have a lot of (empty) extra lines. Better wrap the first line in a `<p>..</p>`, then remove all line breaks (HTML will then take care for the rest).

> There is no need to actually use fastlane e.g. for CI?

Nope, neither for F-Droid nor for my repo (where your app will [show up](https://apt.izzysoft.de/fdroid/index/apk/app.schildpad.schildpad) with the next sync around 6 pm UTC; once it arrives at F-Droid I'd remove it again unless you explicitly want me to keep it). You can use that if you want to deploy to PlayStore that way.

Oh: feel free to keep this issue open if you need it for tracking F-Droid inclusion – or close it as the original request was fulfilled :smiley: Thanks a lot!

**Comment by joelpaulkoch** [comment]
I realized I could simply write the list in markdown style so that's easier for me.

Here is the merge request, there are still some open points for me: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/12830

**Comment by IzzySoft** [comment]
Updated the config at my end to fetch fulldesc along in the future then, thanks! And btw, if you wish you can [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to link to your new presence in my repo :smiley:
